src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -1667,17 +1722,14 @@ public void Warning(string p, bool isPersistent = false)
             if (isPersistent)
             {
                 State = ElementState.PersistentWarning;
-                if (!string.Equals(persistentWarning, p))
-                {
-                    persistentWarning += p;
-                }
+                persistentWarnings.Add(p);
                 ToolTipText = persistentWarning;
             }
             else
             {
                 State = ElementState.Warning;
-                ToolTipText = string.IsNullOrEmpty(persistentWarning) ? p : string.Format("{0}\n{1}", persistentWarning, p);
-                ClearPersistentWarning();
```

Comment:
This would be the biggest change in the Warning logic.
We will not clear persisting warnings when adding a transient warning. (not sure why we ever did).